File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/AddEdgeCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class AddEdgeCommandMsg extends GraphBuildingCommandMsg
{
    Long node1Id;
    Long node2Id;
    Integer outputNumber;
    Integer inputNumber;
    long edgeTemporaryId;

    public AddEdgeCommandMsg(Long node1Id, Long node2Id, Integer outputNumber, Integer inputNumber, long edgeTemporaryId)
    {
        this.node1Id = node1Id;
        this.node2Id = node2Id;
        this.outputNumber = outputNumber;
        this.inputNumber = inputNumber;
        this.edgeTemporaryId = edgeTemporaryId;
    }

    public Long getNode1Id()
    {
        return node1Id;
    }

    public Long getNode2Id()
    {
        return node2Id;
    }

    public Integer getOutputNumber()
    {
        return outputNumber;
    }

    public Integer getInputNumber()
    {
        return inputNumber;
    }

    public long getEdgeTemporaryId()
    {
        return edgeTemporaryId;
    }

}

File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/AddEdgeCommandMsg.js
const GraphBuildingCommandMsg = require('./GraphBuildingCommandMsg.js').GraphBuildingCommandMsg;

class AddEdgeCommandMessage extends GraphBuildingCommandMsg
{
    constructor(node1Id, node2Id, outputNumber, inputNumber, edgeTemporaryId)
    {
        super();
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("AddEdgeCommandMsg");
        this.jsonMessage["node1Id"] = node1Id;
        this.jsonMessage["node2Id"] = node2Id;
        this.jsonMessage["outputNumber"] = outputNumber;
        this.jsonMessage["inputNumber"] = inputNumber;
        this.jsonMessage["edgeTemporaryId"] = edgeTemporaryId;
    }
}

module.exports = {AddEdgeCommandMsg: AddEdgeCommandMessage};

